portal-impl/src/com/liferay/portal/service/impl/AddressLocalServiceImpl.java

```diff
@@ -503,6 +507,29 @@ else if (Validator.isNull(zip)) {
 				listTypeId, classNameId, ListTypeConstants.ADDRESS);
 		}
 
+		if (Validator.isNotNull(subtype) &&
+			(classNameId == _classNameLocalService.getClassNameId(
+				"com.liferay.account.model.AccountEntry.address"))) {
+
+			ListType listType = _listTypeLocalService.getListType(listTypeId);
+
+			String externalReferenceCode =
+				AccountEntryAddressSubtypeConfigurationManagerUtil.
+					getAddressSubtypeListTypeDefinitionExternalReferenceCode(
+						companyId, listType.getName());
+
+			long listTypeEntryId =
+				ListTypeEntryManagerUtil.
+					getListTypeEntryIdByListTypeDefinitionExternalReferenceCode(
+						externalReferenceCode, companyId, subtype);
+
+			if (Validator.isNull(externalReferenceCode) ||
+				(listTypeEntryId == 0)) {
+
+				throw new AddressSubtypeException();
+			}
```

Comment:
please split this if in two (one for the erc and one for the listTypeEntryId) because otheriwse we are passing null to the `getListTypeEntryIdByListTypeDefinitionExternalReferenceCode`